source_test.go

```diff
@@ -60,3 +70,175 @@ func TestSource_Open_WithPosition(t *testing.T) {
 	is.NoErr(err)
 	is.True(pos.LastEditedTime.Equal(underTest.lastMinuteRead))
 }
+
+func TestSource_Configure(t *testing.T) {
+	tests := []struct {
+		desc   string
+		input  map[string]string
+		output Config
+		err    error
+	}{
+		{
+			desc: "Succeed without override",
+			input: map[string]string{
+				Token: "abc-def",
+			},
+			output: Config{token: "abc-def", pollInterval: time.Minute},
+		},
+		{
+			desc: "Succeed with override",
+			input: map[string]string{
+				Token:        "abc-def",
+				PollInterval: "2m",
+			},
+			output: Config{token: "abc-def", pollInterval: 2 * time.Minute},
+		},
+		{
+			desc: "Fail to override poll interval because too small",
+			input: map[string]string{
+				Token:        "abc-def",
+				PollInterval: "1s",
+			},
+			err: fmt.Errorf("poll interval must not be shorter than a minute (provided: 1s)"),
+		},
+		{
+			desc: "Fail to override poll interval because not a time",
+			input: map[string]string{
+				Token:        "abc-def",
+				PollInterval: "a",
+			},
+			err: fmt.Errorf("cannot parse poll interval \"a\": time: invalid duration \"a\""),
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.desc, func(t *testing.T) {
+			ctx := context.Background()
+			s := &Source{}
+
+			err := s.Configure(ctx, tc.input)
+			if tc.err == nil {
+				require.NoError(t, err)
+				assert.Equal(t, tc.output.token, s.config.token)
+				assert.Equal(t, tc.output.pollInterval, s.config.pollInterval)
+			} else {
+				assert.Equal(t, tc.err.Error(), err.Error())
+			}
+		})
+	}
+}
+
+func TestSource_Read(t *testing.T) {
```

Comment:
I like these tests!

An FYI, I'm working on something similar in the refactoring PRs, [here for source_test.go](https://github.com/conduitio-labs/conduit-connector-notion/pull/25/files#diff-5cd78890c07a8996064181cbee424682cf2076232581ed0467aed6509c5e5b4e) and here for [client_test.go](https://github.com/conduitio-labs/conduit-connector-notion/pull/27/files#diff-57cdc3e29d0ef725a50897a4fbd1c4f76353429fa66d3a785dcbc17dcd7b6f31) 

The refactoring adds a small client abstraction, containing only the methods the connector requires. If we continue that path, we'd then put these tests into client_test and not source_test.